ga4gh/backends.py

```diff
@@ -411,20 +416,20 @@ def searchVariantSets(self, request):
         if request.pageSize is None:
             request.pageSize = 100
         response = protocol.GASearchVariantSetsResponse()
-        j = 0
+        currentIndex = 0
         if request.pageToken is not None:
-            j = int(request.pageToken)
-        v = []
-        while j < len(self._variantSetIds) and len(v) < request.pageSize:
-            vs = protocol.GAVariantSet()
-            vs.id = self._variantSetIds[j]
-            vs.datasetId = "NotImplemented"
-            vs.metadata = self._variantSets[vs.id].getMetadata()
-            v.append(vs)
-            j += 1
-        if j < len(self._variantSetIds):
-            response.nextPageToken = j
-        response.variantSets = v
+            currentIndex = int(request.pageToken)
+        variantSetList = []
+        while currentIndex < len(self._variantSetIds) and len(variantSetList) < request.pageSize:
```

Comment:
Yeah, this is cleaner all right. What I'm really keen to do though is abstract away all of this paging logic away from the individual methods. Paging should be implemented exactly once, and I think a nice way to do it is to make  accessor methods (`getVariants`, `getVariantSets`, etc) that are generators over all the objects specified in the request. There are some details about how we deal with pageTokens, but I'm sure we can figure this out. I outlined this idea in the discussion for #10 earlier.

I guess the larger point here is that I don't see much of this code as 'final'; a lot of it needs to be aggressively refactored as we generalise to support the rest of the protocol. As we refactor it we should definitely examine the style and make sure that it's as easy as possible to understand.